File: Fair-Rations.cpp
#include <bits/stdc++.h>

using namespace std;

string ltrim(const string &);
string rtrim(const string &);
vector<string> split(const string &);

/*
 * Complete the 'fairRations' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts INTEGER_ARRAY B as parameter.
 */

string fairRations(vector<int> B)
{
    int lastID = -1;
    int count = 0;

    for (int i = 0; i < B.size(); ++i)
    {
        if (B[i] % 2 != 0)
        {
            if (lastID == -1)
            { 
                lastID = i;
            }
            else
            {
                count += std::abs(i - lastID) * 2;
                lastID = -1;
            }
        }
    }
    
    return (lastID == -1) ? std::to_string(count) : "NO";
}

File: Fair-Rations.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

class Result {

    /*
     * Complete the 'fairRations' function below.
     *
     * The function is expected to return a STRING.
     * The function accepts INTEGER_ARRAY B as parameter.
     */

    public static String fairRations(List<Integer> B) {
    // Write your code here
        int lastID = -1;
        int count = 0;

        for (int i = 0; i < B.size(); ++i) {
            if (B.get(i) % 2 != 0) {
                if (lastID == -1) {
                    lastID = i;
                } else {
                    count += Math.abs(i - lastID) * 2;
                    lastID = -1;
                }
            }
        }

        return (lastID == -1) ? Integer.toString(count) : "NO";
    }

}

public class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        int N = Integer.parseInt(bufferedReader.readLine().trim());

        List<Integer> B = Stream.of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
            .map(Integer::parseInt)
            .collect(toList());

        String result = Result.fairRations(B);

        bufferedWriter.write(result);
        bufferedWriter.newLine();

        bufferedReader.close();
        bufferedWriter.close();
    }
}
